devicetypes/qubino/qubino-flush-dimmer-0-10V.src/qubino-flush-dimmer-0-10V.groovy

```diff
@@ -0,0 +1,671 @@
+        result << createEvent(name: "level", value: cmdValue == 99 ? 100 : cmdValue)
+    }
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd, ep = null) {
+    log.info "SensorMultilevelReport: ${cmd}, endpoint: ${ep}"
+    def result = []
+    def input1SwitchType = Integer.parseInt(state.currentPreferencesState.input1SwitchType.value)
+    log.debug "----------input1SwitchType: ${input1SwitchType}"
+
+    if (input1SwitchType == INPUT_TYPE_GENERAL_PURPOSE_SENSOR && ep == 2) {
+        log.debug "General purpose sensor is handled with 0-100 level values"
+        def roundedVal = Math.round(cmd.scaledSensorValue)
+        createDimmerEvents(roundedVal)
+    } else if (ep == 2)  {
+        def map = [:]
+        switch (cmd.sensorType) {
+            case 1:
+                map.name = "temperature"
+                def cmdScale = cmd.scale == 1 ? "F" : "C"
+                map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+                map.unit = getTemperatureScale()
+                break
+            case 3:
+                map.name = "illuminance"
+                map.value = cmd.scaledSensorValue.toInteger().toString()
+                map.unit = "lux"
+                map.isStateChange = true
+                break;
+            default:
+                map.descriptionText = cmd.toString()
+        }
+        log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+        handleChildEvent(map)
+        result << createEvent(map)
+    }
+}
+
+def handleChildEvent(map) {
+    def childDni = "${device.deviceNetworkId}:" + childDeviceSpecifics[map.name].childDni
+    log.debug "handleChildEvent / find child device: ${childDni}"
+    def childDevice = childDevices.find { it.deviceNetworkId == childDni }
+
+    if(!childDevice) {
+        log.debug "handleChildEvent / creating a child device"
+        def childComponentLabel	= childDeviceSpecifics[map.name].componentLabel
+        def childDthName		= childDeviceSpecifics[map.name].childDthName
+        def childDthNamespace 	= childDeviceSpecifics[map.name].childDthNamespace
+
+        createChildDevice(childDthNamespace.toString(), childDthName.toString(), childDni.toString(), childComponentLabel.toString())
+        childDevice = childDevices.find { it.deviceNetworkId == childDni }
+    }
+    log.debug "handleChildEvent / sending event: ${map} to child: ${childDevice}"
+    childDevice?.sendEvent(map)
+}
+
+def getChildDeviceSpecifics() {
+    [
+        "temperature" : [componentLabel: "Qubino Temperature Sensor", childDni:2, childDthName: "Child Temperature Sensor", childDthNamespace: "qubino"],
+        "illuminance" : [componentLabel: "Qubino Illuminance Sensor", childDni:3, childDthName: "Child Illuminance Sensor", childDthNamespace: "smartthings"]
+    ]
+}
+
+def createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel) {
```

Comment:
the way you're using this, `childDni` is already the parent DNI + the endpoint number